gateway-ha/src/main/java/io/trino/gateway/ha/resource/GatewayWebAppResource.java

```diff
@@ -423,4 +439,71 @@ public Response readExactMatchSourceSelector()
         List<ResourceGroupsManager.ExactSelectorsDetail> selectorsDetailList = resourceGroupsManager.readExactMatchSourceSelector();
         return Response.ok(Result.ok(selectorsDetailList)).build();
     }
+
+    @GET
+    @RolesAllowed("USER")
+    @Produces(MediaType.APPLICATION_JSON)
+    @Path("/getRoutingRules")
+    public Response getRoutingRules()
+    {
+        try {
+            String rulesConfigPath = configuration.getRoutingRules().getRulesConfigPath();
+            YAMLFactory yamlFactory = new YAMLFactory();
+            ObjectMapper yamlReader = new ObjectMapper(yamlFactory);
+            YAMLParser yamlParser = yamlFactory.createParser(new String(Files.readAllBytes(Paths.get(rulesConfigPath))));
+            List<RoutingRules> routingRulesList = yamlReader
+                    .readValues(yamlParser, new TypeReference<RoutingRules>() {})
+                    .readAll();
+            return Response.ok(Result.ok(routingRulesList)).build();
+        }
+        catch (IOException e) {
+            throw new RuntimeException(e);
```

Comment:
Why do we rethrow as RuntimeException? Why not throw IOException directly?

Reply:
My bad, i will fix it.